taxonomy/tests/test_api.py

```diff
@@ -0,0 +1,346 @@
+
+        # pass in None, violates constraints
+        self.assertRaises(IntegrityError,
+                          lambda: create_vocabulary(self.repository.id,
+                                                    "vocabulary",
+                                                    "description",
+                                                    None,
+                                                    "m",
+                                                    9))
+
+        self.assertRaises(TaxonomyAPIException,
+                          lambda: create_vocabulary(self.repository.id,
+                                                    "vocabulary",
+                                                    "description",
+                                                    False,
+                                                    ";",
+                                                    9))
+
+    def test_delete_term(self):
+        """
+        Test delete_term
+        """
+        self.assertEquals(1, Term.objects.count())
+        delete_term(self.term.id)
+        self.assertEquals(0, Term.objects.count())
+        delete_term(9)
+        delete_term(self.term.id)
+        self.assertEquals(0, Term.objects.count())
+
+    def test_delete_vocabulary(self):
+        """
+        Test delete_vocabulary
+        """
+        self.assertEquals(1, Vocabulary.objects.count())
+        delete_vocabulary(9)
+        self.assertEquals(1, Vocabulary.objects.count())
+        delete_vocabulary(self.vocabulary.id)
+        self.assertEquals(0, Vocabulary.objects.count())
+
+    def test_get_learning_resources_for_term(self):
+        """
+        Test get_learning_resources_for_term
+        """
+        self.assertEquals([],
+                          list(get_learning_resources_for_term(self.term.id)))
+
+        self.learning_resource.terms.add(self.term)
+        self.assertEquals([self.learning_resource],
+                          list(get_learning_resources_for_term(self.term.id)))
+
+        self.assertRaises(Term.DoesNotExist,
+                          lambda: get_learning_resources_for_term(9))
+
+    def test_get_term(self):
+        """
+        Test get_term
+        """
+        self.assertRaises(Term.DoesNotExist, lambda: get_term(9))
+        self.assertEquals(self.term, get_term(self.term.id))
+
+    def test_get_terms_for_learning_resource(self):
+        """
+        Test get_terms_for_learning_resource
+        """
+        self.assertEquals([],
+                          list(get_terms_for_learning_resource(
+                              self.learning_resource.id)))
+
+        self.term.learning_resources.add(self.learning_resource)
+        self.assertEquals([self.term],
+                          list(get_terms_for_learning_resource(
+                              self.learning_resource.id)))
+        self.assertRaises(LearningResource.DoesNotExist,
+                          lambda: get_terms_for_learning_resource(9))
```

Comment:
The "magic numbers" used in these tests are distracting. Maybe use `0` or `-1` in all places (instead of sometimes `3`, sometimes `9`), or set `FAKE_ID = -1` at the top and use it everywhere.